### ISSUE TITLE
Cuda finds only one device when num_gpus > 1

### ISSUE THREAD

**Issue by npyoung** [ray-project/ray#3198]
### System information
- **OS Platform and Distribution**: Ubuntu 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.5.3
- **Python version**: 3.6
- **Exact command to reproduce**:

```python
import os
import ray
import pycuda.driver as cuda
cuda.init()
print(cuda.Device.count()) # prints "6"

ray.init(num_gpus=6, use_raylet=True)

@ray.remote(num_gpus=4)
def test_multigpu():
    import pycuda.driver as cuda
    cuda.init()
    return cuda.Device.count()

print(ray.get(test_multigpu.remote())) # prints "1"
```
### The problem
Ray remote functions only give pycuda access to a single GPU even when `num_gpus` is set >1 on a machine with >1 GPU. Not only does the device count report that there is only one GPU, but if you attempt to create contexts on GPUs with ordinal >0, context creation fails.

`os.environ["CUDA_VISIBLE_DEVICES"]` inside the remote function does in fact give a list of `num_gpus` comma-separated values as expected.

**Comment by richardliaw** [comment]
OK oddly enough, I ran your script on our local machine:

```
In [3]: import os
   ...: import ray
   ...: import pycuda.driver as cuda
   ...: cuda.init()
   ...: print(cuda.Device.count()) # prints "6"
   ...:
   ...: ray.init(num_gpus=6, use_raylet=True)
   ...:
   ...: @ray.remote(num_gpus=4)
   ...: def test_multigpu():
   ...:     import pycuda.driver as cuda
   ...:     cuda.init()
   ...:     return cuda.Device.count()
   ...:
   ...: print(ray.get(test_multigpu.remote())) # prints "1"
8
Process STDOUT and STDERR is being redirected to /tmp/raylogs/.
Waiting for redis server at 127.0.0.1:23597 to respond...
Waiting for redis server at 127.0.0.1:59449 to respond...
Starting the Plasma object store with 216.00 GB memory.
Failed to start the UI, you may need to run 'pip install jupyter'.
WARNING: Logging before InitGoogleLogging() is written to STDERR
E1102 16:00:19.041138 77122 io.cc:119] Connection to socket failed for pathname /tmp/raylet91997958
4
```

**Comment by npyoung** [comment]
This was a case where the issue could be captured well by a short piece of code.

I tried running this script again on an EC2 p3 instance and got the correct number of GPUs in the second print statement. I discovered this issue on an older machine with older GPUs and CUDA 8. Might be a CUDA issue or card issue.

I still did not get my application working on EC2 and discovered a similar, but different problem. It's definitely a more specific issue (and definitely caused by something ray is doing) so I'll open a new issue and close this one.
